io/net/virt-net/network_virtualization.py

```diff
@@ -594,6 +597,59 @@ def test_vnic_hmc_dlpar(self):
                 count += 1
         self.check_dmesg_error()
 
+    def test_vnic_eeh(self):
+        """
+        Perform EEH on vnic interface from vios
+        """
+        current_logport = self.get_active_device_logport(self.slot_num[0])
+        if not self.original_logport == current_logport:
+            self.trigger_failover(self.original_logport)
+        else:
+            self.log.info("Unable to set the logport to original one")
+        time.sleep(5)
+        self.session = Session(self.vios_ip, user=self.vios_user,
+                               password=self.vios_pwd)
+        self.session.cleanup_master()
+        if not wait.wait_for(self.session.connect, timeout=30):
+            self.fail("Failed connecting to VIOS")
+        cmd = "ioscli lsmap -all -vnic -cpid %s" % self.lpar_id
+        vnic_servers = self.session.cmd(cmd).stdout_text.splitlines()
+        device = self.find_device(self.mac_id[0])
+        temp_idx = vnic_servers.index("Client device name:" + device)
+        vnic_bankingdevice = vnic_servers[temp_idx - 3].split(":")[1]
+        s = pxssh.pxssh()
+        try:
+            s.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        except Exception:
+            self.warn("Unable to login to vios")
+        time.sleep(2)
+        s.sendline("oem_setup_env")
+        time.sleep(5)
+        s.sendline("kdb")
+        time.sleep(5)
+        s.sendline("set scroll false")
+        time.sleep(5)
+        cmd = "mlxcent setacs %s" % vnic_bankingdevice
+        s.sendline(cmd)
+        time.sleep(2)
+        s.sendline("mlxcent pollq 0")
+        s.prompt()
+        mapstart = s.before.decode("utf-8").split("\r\n ")
+        for i in mapstart:
+            if re.search("map_start", i):
+                map_start_value = i.split("=")[1]
+        s.sendline("quit")
+        cmd = "./eeh_tool_64 %s 3 15 -w 64 -a %s -m 0xFFFFFFFFFFFFF000" % (vnic_bankingdevice, map_start_value)
```

Comment:
make sure the eeh_tool is available in .py..data folder of this test.. and add a logic to copy every time to vios

Reply:
can you create a scp command by getting the local lpar ip and path to binary  and run on vios

cmd = "scp lparuser@lparip:/path/to/data/file/binary ." % (lpar_user, lpar_ip, bin_path)
sendline(cmd)
expect(passwd:)
sendline(lpar_pass)